runtime/lua/vim/lsp/util.lua

```diff
@@ -1290,18 +1290,169 @@ function M._trim(contents, opts)
   return contents
 end
 
---- Generates a table mapping markdown code block lang to vim syntax,
---- based on g:markdown_fenced_languages
----@return table table of lang -> syntax mappings
-local function get_markdown_fences()
-  local fences = {}
-  for _, fence in pairs(vim.g.markdown_fenced_languages or {}) do
-    local lang, syntax = fence:match('^(.*)=(.*)$')
-    if lang then
-      fences[lang] = syntax
-    end
-  end
-  return fences
+---@alias MarkdownBlock {line:string}
+---@alias MarkdownCodeBlock {code:string[], lang:string}
+---@alias Markdown (MarkdownBlock|MarkdownCodeBlock)[]
+
+---@param text string
+---@param opts? {ft?: string}
+---@return Markdown
+local function parse_markdown(text, opts)
+  opts = opts or {}
+  text = text:gsub('</?pre>', '```'):gsub('\r', '')
```

Comment:
Is it possible to fix this: https://github.com/folke/noice.nvim/issues/424?

I have been using the diff from my comment: https://github.com/folke/noice.nvim/issues/424#issuecomment-1510268975 ever since I posted the comment, and can't remember having any issues.

I am not sure if that is the best way to fix it, and some others probably know this better than I do, but for me it has worked perfectly.